script/run-review-instance-api.js

```diff
@@ -0,0 +1,123 @@
+      const reviewInstanceDeployment = data.find(
+        deployment => deployment.description !== 'Heroku',
+      );
+
+      if (!reviewInstanceDeployment) {
+        throw new Error(
+          "No review instance deployment activity found. Deploy probably hasn't kicked off yet.",
+        );
+      }
+
+      return reviewInstanceDeployment;
+    });
+};
+
+// Get the object repreesenting the successful review instance deployment.
+// It won't exist yet if the Jenkins build is still in progress.
+const getSuccessfulDeployment = deploymentData => {
+  // Set this header to go through the GitHub preview API so we can access the
+  // `environment_url` attribute later.
+  const options = {
+    headers: {
+      Accept: 'application/vnd.github.ant-man-preview+json',
+    },
+  };
+
+  return fetch(deploymentData.statuses_url, options)
+    .then(response => response.json())
+    .then(data => {
+      if (!data.length) {
+        throw new Error(
+          "No review instance deployment statuses found. Deploy probably hasn't kicked off yet.",
+        );
+      }
+
+      // Response should be an array of statuses representing the activity for
+      // this deployment. Look for a deployment status tagged as 'success' (as
+      // opposed to just 'pending').
+      const successfulDeployment = data.find(
+        deployment => deployment.state === 'success',
+      );
+
+      if (!successfulDeployment) {
+        throw new Error(
+          'No successful review instance deployment found. Review instance might still be getting ready.',
+        );
+      }
+
+      return successfulDeployment;
+    });
+};
+
+// Get the URL where the review instance was deployed.
+// This is the same URL as the 'View deployment' link in the PR.
+const getEnvironmentUrl = successfulDeployment => {
+  const { environment_url } = successfulDeployment;
+
+  if (!environment_url) {
+    throw new Error(
+      "Couldn't find the review instance URL. The environment_url property might no longer be included with the GitHub preview API, or it might not have been specified with the deploy.",
+    );
+  }
+
+  return environment_url;
+};
+
+// Build the review instance API URL based on the environment URL.
+// Just append '-api' to the first segment of the URL.
+const buildReviewInstanceApiUrl = environmentUrl =>
+  environmentUrl
+    .split('.')
+    .reduce(
+      (acc, curr, idx) => (idx === 0 ? `${acc}${curr}-api` : `${acc}.${curr}`),
+      '',
+    );
+
+// Run the watch task with the build pointing to the review instance API.
+const runWatch = apiUrl =>
+  execSync(`yarn watch --api=${apiUrl}`, { stdio: 'inherit' });
+
+const logError = error => {
+  console.error(error);
+  return Promise.reject(error);
+};
+
+getCommitHash()
+  .then(getDeploymentData)
+  .then(getSuccessfulDeployment)
+  .then(getEnvironmentUrl)
+  .then(buildReviewInstanceApiUrl)
+  .catch(logError)
+  .then(runWatch)
```

Comment:
minor nit but maybe group the `catch` statements at the end. I don't think this `then` will be hit when `logError` returns a rejected promise.

Reply:
Ah, so this was on purpose... I noticed that if the `yarn watch` task itself had any errors (in my case, it was the thing where the Drupal cache needed to get updated), the error being logged with `logError` wasn't any more helpful than what was already being logged with `yarn watch`.

Keeping the `catch(logError)` at the end would append some object specific to the `execSync` call with some internal properties, which didn't seem that relevant and would likely just confuse developers.

Reply:
If any of the prior steps failed, we shouldn't run the `yarn watch` anyway, since it would end up with something like `yarn watch --api="undefined"`.